plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/configuration/BaseVersionedStorageFormat.java

```diff
@@ -27,13 +28,23 @@ public enum BaseVersionedStorageFormat implements VersionedStorageFormat {
    * make BlobDB more effective
    */
   FOREST_WITH_VARIABLES(DataStorageFormat.FOREST, 2),
+  /**
+   * Current Forest version, with receipts using compaction, in order to make Receipts use less disk
+   * space
+   */
+  FOREST_WITH_RECEIPT_COMPACTION(DataStorageFormat.FOREST, 3),
```

Comment:
Didn't add this for private transactions as private transactions don't store the bloom filter and most of the storage savings come from not storing the bloom filter. Will run a test and see what benefits in storage we get from just the log encoding improvements, maybe it will be worth adding to the private transactions.

Reply:
I am referring to the public storage of the privacy mode, if `PrivacyVersionedStorageFormat` is not extended to support the new v3 as an option it will stay on v2

Reply:
I get it now. Didn't release the versioned storage format worked that way. Nice catch. I've updated the `PrivacyVersionedStorageFormat` so that if privacy is enabled then the version format will also be updated when receipt compaction is enabled.